src/logic/contracts/safeContracts.ts

```diff
@@ -28,68 +31,91 @@ let safeMaster: GnosisSafe
  * @param {Web3} web3
  * @param {ETHEREUM_NETWORK} networkId
  */
-const createGnosisSafeContract = (web3: Web3, networkId: ETHEREUM_NETWORK) => {
+export const getGnosisSafeContract = memoize((networkId: ETHEREUM_NETWORK) => {
```

Comment:
Check if this memoize is working ok maybe the problem is here

Reply:
it's working fine